src/selectedtests/test_mappings/update_test_mappings.py

```diff
@@ -11,6 +14,52 @@
 LOGGER = structlog.get_logger()
 
 
+def update_test_mappings(test_mappings: List[Dict], mongo: MongoWrapper) -> None:
+    """
+    Update test mappings in the test mappings project config collection.
+
+    :param test_mappings: A list of test mappings.
+    :param mongo: An instance of MongoWrapper.
+    """
+    for mapping in test_mappings:
+        # get the test_files to update and remove from the mapping document.
+        test_files = mapping.get("test_files", [])
+        del mapping["test_files"]
+
+        # get the value to inc  and remove from the mapping document (otherwise we would double
+        # increment).
+        source_file_seen_count = mapping["source_file_seen_count"]
+        del mapping["source_file_seen_count"]
+
+        # we don't need this in the $set as it is supplied by query.
+        source_file = mapping["source_file"]
+        del mapping["source_file"]
+
+        result = mongo.test_mappings().update_one(
+            {"source_file": source_file},
+            {"$set": mapping, "$inc": {"source_file_seen_count": source_file_seen_count}},
```

Comment:
Would it be reasonable to `$set` something explicitly with values from `mapping`?
e.g.:
```
"$set": {
    "something": mapping["something"],
    "another_thing": mapping["another_thing"]
}
```

It seems like this would be easier to reason about given that this function just knows that `mapping` is some kind of dictionary, and remove the need to delete entries above.